### DISCUSSION TITLE
Solcon IPTV not working

### DISCUSSION THREAD

**Discussion by JHAvB** [fabianishere/udm-iptv#6] [Q&A] [answered]
It seems that everything works until the UDM.
 
Only the IPTV decoder is not coming up automatically on the right network, when I force this it seems that the IPTV decoder is connected to the internet instead of the IPTV network. 
I have absolutely no idea what is going wrong. Hopefully someone can help me out.

Internet does not need a VLAN and goes via DHCP
IPTV VLAN 188
Route to 10.0.0.0/8
IP Range 10.232.0.0/16 10.233.0.0/16

15-iptvs.sh:

IPTV_WAN_INTERFACE="eth4"
IPTV_WAN_RANGES="10.252.0.0/16 10.253.0.0/16"
IPTV_WAN_VLAN="188"
IPTV_WAN_DHCP_OPTIONS="-O staticroutes -V IPTV_RG"
IPTV_LAN_INTERFACES="br50"
IPTV_LAN_RANGES=""

if podman container exists iptv; then
  podman rm -f iptv
fi
podman run --network=host --privileged \
    --name iptv -i -d --restart on-failure:5 \
    -e IPTV_WAN_INTERFACE="$IPTV_WAN_INTERFACE" \
    -e IPTV_WAN_RANGES="$IPTV_WAN_RANGES" \
    -e IPTV_WAN_VLAN="$IPTV_WAN_VLAN" \
    -e IPTV_WAN_DHCP_OPTIONS="$IPTV_WAN_DHCP_OPTIONS" \
    -e IPTV_LAN_INTERFACES="$IPTV_LAN_INTERFACES" \
    -e IPTV_LAN_RANGES="$IPTV_LAN_RANGES" \
    fabianishere/udm-iptv -d -
![Afbeelding1](https://user-images.githubusercontent.com/92083215/137143464-66b35f64-8b06-4a90-8f9a-68ba1be41a2d.png)
![Afbeelding2](https://user-images.githubusercontent.com/92083215/137143470-c9279aaf-3457-4703-ba3a-392139f902ee.png)
![Afbeelding3](https://user-images.githubusercontent.com/92083215/137143472-c7670698-b90b-40b0-b9c7-a6af266f1743.png)
v

**Comment by fabianishere** [comment]
Could you clarify what you mean by
> Only the IPTV decoder is not coming up automatically on the right network, when I force this it seems that the IPTV decoder is connected to the internet instead of the IPTV network.

Do you mean that the IPTV decoder is not getting an IP address from your IPTV VLAN?

**Reply by JHAvB** [reply]
correct, the IPTV receiver does not get the correct IP address, if I force this through the port on IPTV vlan then it does. 

Only when booting the IPTV receiver we get "boot argument not found" which indicates that the IPTV receiver is connected to the internet and not to the IPTV on VLAN 188. 

Option 60 IPTV_RG is on

**Reply by fabianishere** [reply]
It is probably necessary to set the profile of the switch port connected to the IPTV receiver to the IPTV VLAN, since I don't expect the IPTV receiver is VLAN-aware.

I see you have set `IPTV_LAN_INTERFACES` to `br50`, but I don't see any route for this interface is your route list. Are you sure this is the correct interface? I do see you have a `br188` bridge, which is suspect is LAN (with VLAN ID 188). Is that your internal IPTV VLAN? If so, you should set `IPTV_LAN_INTERFACES` to `br188`.

**Reply by JHAvB** [reply]
![Schermafbeelding 2021-10-13 155816](https://user-images.githubusercontent.com/92083215/137147542-bfaf6b9e-d9a0-43f1-81b7-812a01b2643c.png)


yes sorry, I was testing with the VLAN

if indeed I set the port profile to IPTV VLAN, I get the right IP but still not the right connection. It still seems that the IPTV receiver is connected to the internet

**Reply by fabianishere** [reply]
It is expected that you still have access to the internet from your IPTV VLAN (or for that matter all your VLANs unless you explicitly disallow it through firewall rules). The container just makes sure that IPTV multicast packets get routed correctly between your receiver and the IPTV network.

**Reply by JHAvB** [reply]
so basically it runs aground on the boot argument that the IPTV cannot receive

**Reply by fabianishere** [reply]
It might be that you are missing a NAT/firewall rule. Are you sure the WAN ranges are correct?

**Reply by fabianishere** [reply]
See this topic: https://community.ui.com/questions/SOLVED-Solcon-internet-and-IPTV-working-via-Unifi-USG3p-and-replacing-the-Fritzbox-7530/6160468a-1649-416c-9753-b0fe0747f292?page=1

Apparently, the IPTV receiver expects certain DHCP options that are provided by the WAN DHCP server. You’ll need to sniff these values and manually set them for your internal DHCP server.

**Reply by JHAvB** [reply]
I received the following from ISP:
you need to instal a route to 10.0.0.0/8 and need to receive an IP via DHCP in range 10.252.0.0/16 or 10.253.0.0/16 based on the vendor-class-identifier: IPTV_RG then it’s possible to use IPTV without the Fritsbox. 

I requested the above settings again from my ISP. 

Currently, the IPTV-receiver is booted but we the screen remains black. However, it is possible to use the interactive tv.

**Comment by JHAvB** [comment]
Thank you very much for your help and at all for this page!!!

Via the link I now find out that Solcon is using the KPN IPTV server. WAN Ranges now set to 213.75.0.0/16 217.166.0.0/16 and now it works.

Only issue I am still running into is that IPTV network is forced to the receiver. Not really a problem but this is something I will look into later. 

Again thank you very much

**Comment by JHAvB** [comment, marked as answer]
Conclusion for Solcon:


# vim 15-iptv.sh
IPTV_WAN_INTERFACE="eth4"
IPTV_WAN_RANGES="213.75.0.0/16 217.166.0.0/16"
IPTV_WAN_VLAN="188"
IPTV_WAN_DHCP_OPTIONS="-O staticroutes -V IPTV_RG"
IPTV_LAN_INTERFACES="br50"

if podman container exists iptv; then
  podman rm -f iptv
fi
podman run --network=host --privileged \
    --name iptv -i -d --restart on-failure:5 \
    -e IPTV_WAN_INTERFACE="$IPTV_WAN_INTERFACE" \
    -e IPTV_WAN_RANGES="$IPTV_WAN_RANGES" \
    -e IPTV_WAN_VLAN="$IPTV_WAN_VLAN" \
    -e IPTV_WAN_DHCP_OPTIONS="$IPTV_WAN_DHCP_OPTIONS" \
    -e IPTV_LAN_INTERFACES="$IPTV_LAN_INTERFACES" \
    -e IPTV_LAN_RANGES="192.168.50.0/24" \
    fabianishere/udm-iptv -d -v
